src/storage/v2/disk/storage.cpp

```diff
@@ -575,7 +623,26 @@ VerticesIterable DiskStorage::DiskAccessor::Vertices(LabelId label, PropertyId p
           indexed_vertices->access());
     }
   }
+  return gids;
+}
+
+void DiskStorage::DiskAccessor::LoadVerticesFromDiskLabelPropertyIndexForIntervalSearch(
+    LabelId label, PropertyId property, std::unordered_set<storage::Gid> gids,
+    const std::optional<utils::Bound<PropertyValue>> &lower_bound,
+    const std::optional<utils::Bound<PropertyValue>> &upper_bound, std::list<Delta> &index_deltas,
+    utils::SkipList<Vertex> *indexed_vertices) {
+  auto *disk_label_property_index =
+      static_cast<DiskLabelPropertyIndex *>(storage_->indices_.label_property_index_.get());
+
+  auto disk_index_transaction = disk_label_property_index->CreateRocksDBTransaction();
+  disk_index_transaction->SetReadTimestampForValidation(transaction_.start_timestamp);
+  rocksdb::ReadOptions ro;
+  std::string strTs = utils::StringTimestamp(transaction_.start_timestamp);
+  rocksdb::Slice ts(strTs);
+  ro.timestamp = &ts;
+  auto index_it = std::unique_ptr<rocksdb::Iterator>(disk_index_transaction->GetIterator(ro));
```

Comment:
Is it possible to split this function into more functions, so 1 would be LoadVerticesFromDiskLabelPropIndex and 2. would be do interval search?

Reply:
Maybe up until here we can call it Load... or CreateIterator, and afterwards it is DoIntervalSearch...

Reply:
I mean we can but then I have to check every such function.. And the reason I didn't do it from the beginning is because ro.timestamp needs pointer so I cannot allocate timestamp on stack